src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -1007,7 +1167,7 @@ public function has_class( $wanted_class ) {
 	 */
 	public function set_bookmark( $name ) {
 		// It only makes sense to set a bookmark if the parser has paused on a concrete token.
```

Comment:
This comment confused me with the change below. Would this read better?

```suggestion
		// A bookmark cannot be set if we've reached the end of the document.
```

Reply:
It could be the case that the parser hasn't started yet, or that it has been extended after reaching a partial ending (not yet implemented, but planned).

In these cases I was trying to draw the positive assertion, that bookmarking can only bookmark matched tokens and is senseless to attempt when the parser hasn't identifying a particular thing.

Happy to try and clarify further, but the rule applies in cases not limited to reaching the end of the document.